### ISSUE TITLE
family driver not working

### ISSUE THREAD

**Issue by source99** [h5py/h5py#920]
I consistently get this error when trying to create a new hdf5 file or open an existing hdf5 file when using the family driver.

OSError: Unable to create file (file names not unique)

Trying to create a new file:
(tdd) matt@Matt-laptop src (master) $ python
Python 3.5.2 (v3.5.2:4def2a2901a5, Jun 26 2016, 10:47:25) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import h5py
>>> new_file = h5py.File("test_new_filename.hdf5", "w", driver="family")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/matt/work/centosette/source/tdd/lib/python3.5/site-packages/h5py/_hl/files.py", line 269, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/Users/matt/work/centosette/source/tdd/lib/python3.5/site-packages/h5py/_hl/files.py", line 105, in make_fid
    fid = h5f.create(name, h5f.ACC_TRUNC, fapl=fapl, fcpl=fcpl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5f.pyx", line 98, in h5py.h5f.create
OSError: Unable to create file (file names not unique)

Trying to open an existing file that can be opened with HDFView 2.14
(tdd) matt@Matt-laptop src (master) $ python
Python 3.5.2 (v3.5.2:4def2a2901a5, Jun 26 2016, 10:47:25) 
[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import h5py
>>> filename = "/Users/matt/test0.hdf5"
>>> existing_file = h5py.File(filename, "r", driver="family")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/matt/work/centosette/source/tdd/lib/python3.5/site-packages/h5py/_hl/files.py", line 269, in __init__
    fid = make_fid(name, mode, userblock_size, fapl, swmr=swmr)
  File "/Users/matt/work/centosette/source/tdd/lib/python3.5/site-packages/h5py/_hl/files.py", line 99, in make_fid
    fid = h5f.open(name, flags, fapl=fapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5f.pyx", line 78, in h5py.h5f.open
OSError: Unable to open file (file names not unique)

 * OS=10.12.6 and window 7
 * python = 3.5.2 or 3.6

 * h5py version 2.7.1 but i am able to reproduce the issue in earlier versions also
 * HDF5 version (e.g. 1.8.18)
 * The full traceback/stack trace shown (if it appears)

>>> h5py.version.info
'Summary of the h5py configuration\n---------------------------------\n\nh5py    2.7.1\nHDF5    1.8.18\nPython  3.5.2 (v3.5.2:4def2a2901a5, Jun 26 2016, 10:47:25) \n[GCC 4.2.1 (Apple Inc. build 5666) (dot 3)]\nsys.platform    darwin\nsys.maxsize     9223372036854775807\nnumpy   1.13.1\n'

**Comment by source99** [comment]
This was user error.  I had to name the file filename0.hdf5 and then specify the filename as 'filename_%d.hdf5' in my code.

This does bring up 2 other minor issues - 
1.  There is no testing for 'family' driver in the tests.  I have cloned the repo but I don't know how to run all the tests.  If someone can tell me how to run the tests I can fix it and file a merge.

2.  The help(h5py.File) documentation doesn't list 'family' as a legal driver.  Again if someone points me how to run the tests I can file a merge.
